### PR TITLE
[REEF-1308] Enhance Tang duplicated named parameter entries error message

### PR DESCRIPTION
**Pull request by jwang98052**
* Update COnfigurationBuilderImpl
* Add test case

JIRA: [REEF-1308](https://issues.apache.org/jira/browse/REEF-1308)

This closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Tang/Implementations/Configuration/ConfigurationBuilderImpl.cs [hunk 1]
```diff
@@ -272,7 +272,15 @@ public void BindParameter(INamedParameterNode name, string value)
...
             else 
             {
-                NamedParameters.Add(name, value);
+                try
+                {
+                    NamedParameters.Add(name, value);
+                }
+                catch (ArgumentException e)
+                {
+                    var msg = string.Format(CultureInfo.CurrentCulture, "Duplicated entries: try to bind [{0}] to [{1}], but the configuration has been set for it.", value, name.GetFullName());
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang/Implementations/Configuration/ConfigurationBuilderImpl.cs]: Why `CurrentCulture`?

File: lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    .BindStringNamedParam<NamedString>("abc")
+                    .BindStringNamedParam<NamedString>("def")
+                    .Build();
+            }
+            catch (ArgumentException e)
+            {
+                msg = e.Message;
+            }
+            Assert.NotNull(msg);
+            Assert.True(msg.Contains("An entry with the same key already exists"));
```
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs]: Can you check for the message you're adding instead? This depends on what message the {{Map}} implementation throws.

**Comment by afchung** [comment]
@jwang98052 done with a first pass.

File: lang/cs/Org.Apache.REEF.Tang/Implementations/Configuration/ConfigurationBuilderImpl.cs (hunk 1 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang/Implementations/Configuration/ConfigurationBuilderImpl.cs]: Changed it to InvariantCulture

File: lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs (hunk 2 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs]: Yes. I  checked. This message comes from .Net exception. The message I added is for log purpose only.

**Comment by jwang98052** [comment]
@afchung I have addressed your review comments.

File: lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs (hunk 2 above)
Comment (afchung) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs]: @jwang98052 Right, I'm saying that the behavior of our tests should not depend on the hardcoded Exception message from another library. We should check our own messages instead, if we really want to verify messages.

Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestNegativeCases.cs]: make sense.

**Comment by jwang98052** [comment]
@afchung I have addressed your review comments for exception message validation.

**Comment by afchung** [comment]
@jwang98052 can you rebase and resolve the conflicts? After that, I'll test and merge. Thanks!

**Comment by jwang98052** [comment]
@afchung Rebased.

**Comment by afchung** [comment]
@jwang98052 I'll test and merge. Thanks!

**Pull request closed without merging**